srml/support/procedural/src/storage/impls.rs

```diff
@@ -193,11 +193,10 @@ impl<'a> Impls<'a> {
 		let head_key = format!("head of {}", prefix);
 		let prefix = format!("{}", prefix);
 		let name_lowercase = name.to_string().to_lowercase();
-		let key_for = syn::Ident::new(&format!("key_for_{}", name_lowercase), name.span());
-		let internal_module = syn::Ident::new(&format!("__internal_do_not_use_{}", name_lowercase), name.span());
-		let linkage = syn::Ident::new(&format!("Linkage{}", name), name.span());
-		let borrowing_linkage = syn::Ident::new(&format!("Borrowing{}", linkage), name.span());
-		let enumerator = syn::Ident::new(&format!("Enumerator{}", name), name.span());
+		let key_for = syn::Ident::new(&format!("__decl_storage_key_for_{}", name_lowercase), name.span());
+		let linkage = syn::Ident::new(&format!("__DeclStorageLinkage{}", name), name.span());
+		let enumerator = syn::Ident::new(&format!("__DeclStorageEnumerator{}", name), name.span());
+		let phantom_data = quote! { #scrate::storage::generator::PhantomData };
 		let put_or_insert = quote! {
 			match linkage {
 				Some(linkage) => storage.put(key_for, &(val, linkage)),
```

Comment:
Do we agree this is for optimisation ? I mean we could have done the insert like normal this would have been computed inside insert method isn't it ?

Reply:
Yeah, using `insert` is fine, the only cost is that we read the storage twice for every `mutate`. IMHO the maintance cost of the optimisation is not super high and I think mutating an existing entry is actually a dominant use case so it's worth to optimize.